api/src/opentrons/protocol_engine/state/modules.py

```diff
@@ -375,13 +391,17 @@ def select_hardware_module_to_load(
             ModuleAlreadyPresentError: A module of a different type is already
                 assigned to the requested location.
         """
-        existing_mod = self._state.hardware_module_by_slot.get(location.slotName)
+        existing_mod_in_slot = None
+
+        for mod_id, slot in self._state.slot_by_module_id.items():
+            if slot == location.slotName:
+                existing_mod_in_slot = self._state.hardware_by_module_id.get(mod_id)
```

Comment:
Not that it really matters for performance, but maybe helpful for clarity, we can avoid some unnecessary iteration by:

```suggestion
        for mod_id, slot in self._state.slot_by_module_id.items():
            if slot == location.slotName:
                existing_mod_in_slot = self._state.hardware_by_module_id.get(mod_id)
                break
```

Or move this to a helper function and use an early `return` instead of `break`.